Reorganize example documentation (#11757)
bytecodealliance/wasmtime: docs/lang.md

features. This is not intentional, however, and with sufficient development
resources the two APIs will be kept in-sync.

Documentation for the C API can be found at
[docs.wasmtime.dev/c-api](https://docs.wasmtime.dev/c-api/). Documentation on
how to build the C API can be found in the [README of the C API]. Building the C
API uses CMake to orchestrate the build and Cargo under the hood to build the
Rust code.

The C API can also be installed through `*-c-api-*` [release artifacts].

[README of the C API]: https://github.com/bytecodealliance/wasmtime/blob/main/crates/c-api/README.md
[release artifacts]: github.com/bytecodealliance/wasmtime/releases/latest

## C++

Wasmtime supports C++ as a header-based library layered on top of the C API. The
C++ API header use the `*.hh` extension and are installed alongside the C API
meaning that if you install the C API you've got the C++ API as well. The C++
API is focused on automating resource management of the C API and providing an
easier-to-use API on top.

## Bash

Wasmtime is available in Bash through the [`wasmtime` CLI](./cli.md) executable
and its various subcommands. Source for the `wasmtime` executable is developed

> int he same repository as Wasmtime itself. Installation of `wasmtime` can be

Reviewer: ```suggestion
in the same repository as Wasmtime itself. Installation of `wasmtime` can be
```

[found in this documentation](./cli-install.md).

# External bindings to Wasmtime

The following language bindings are all developed outside of the Wasmtime
repository. Consequently they are not officially supported and may have varying
levels of support and activity. Note that many of these bindings are in the

> bytecodealliance GitHub organization but are still note tested/developed in-sync

Reviewer: ```suggestion
bytecodealliance GitHub organization but are still not tested/developed in-sync
```
